backend/src/main/java/hanglog/expense/service/ExpenseService.java

```diff
@@ -0,0 +1,104 @@
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class ExpenseService {
+
+    //  TODO: 추후 Currency 데이터 생길시 deafault 값 추가
+    private static final Currency DEFAULT_CURRENCY = Currency.getDefaultCurrency();
+
+    private final TripRepository tripRepository;
+    private final CurrencyRepository currencyRepository;
+    private final TripCityRepository tripCityRepository;
+
+    public TripExpenseResponse getAllExpenses(final Long tripId) {
+        final Trip trip = tripRepository.findById(tripId)
+                .orElseThrow(() -> new BadRequestException(NOT_FOUND_TRIP_ID));
+        final Currency currency = currencyRepository.findByDate(trip.getStartDate())
+                .orElse(DEFAULT_CURRENCY);
+
+        final Map<DayLog, Integer> dayLogAmounts = new HashMap<>();
+        final Map<Category, Integer> categoryAmounts = new HashMap<>();
+        final List<TripCity> tripCities = tripCityRepository.findByTripId(tripId);
+
+        for (final DayLog dayLog : trip.getDayLogs()) {
+            calculateAmounts(dayLog, currency, dayLogAmounts, categoryAmounts);
+        }
+
+        final int totalAmount = calculateTotalAmount(dayLogAmounts);
+
+        final List<CategoryExpense> categoryExpenses = categoryAmounts.entrySet().stream()
+                .map(entry -> new CategoryExpense(entry.getKey(), entry.getValue(), totalAmount))
+                .toList();
+
+        final List<DayLogExpense> dayLogExpenses = dayLogAmounts.entrySet().stream()
+                .map(entry -> new DayLogExpense(entry.getKey(), entry.getValue()))
+                .toList();
+
+        return TripExpenseResponse.of(
+                trip,
+                totalAmount,
+                tripCities,
+                categoryExpenses,
+                currency,
+                dayLogExpenses
+        );
+    }
+
+    private void calculateAmounts(
+            final DayLog dayLog,
+            final Currency currency,
+            final Map<DayLog, Integer> dayLogAmounts,
+            final Map<Category, Integer> categoryAmounts
+    ) {
+
+        for (final Item item : dayLog.getItems()) {
+            final Optional<Expense> expense = Optional.ofNullable(item.getExpense());
+
+            if (expense.isPresent()) {
+                final int dayLogAmount = dayLogAmounts.getOrDefault(dayLog, 0);
+                dayLogAmounts.put(dayLog, dayLogAmount + changeToKRW(expense.get(), currency));
+
+                final int categoryAmount = categoryAmounts.getOrDefault(expense.get().getCategory(), 0);
+                categoryAmounts.put(expense.get().getCategory(), categoryAmount + changeToKRW(expense.get(), currency));
+            }
+        }
+    }
+
+    private int changeToKRW(final Expense expense, final Currency currency) {
+        final double rate = CurrencyType.mappingCurrency(expense.getCurrency(), currency);
+        return (int) (expense.getAmount() * rate);
```

Comment:
```suggestion
        return Double.valueOf(expense.getAmount() * rate).intValue();
```

요런 식으로도 가능할 것 같아요! 
다른 로직에서도 타입 캐스팅을 하는게 있던데 달리가 보고 괜찮으면 일관성있게 변경해 주세용